src/domain/service/timer/TimerService.ts

```diff
@@ -0,0 +1,44 @@
+export default class TimerService {
+  private timers: Map<string, { timeoutId: NodeJS.Timeout, startTime: number, duration: number }> = new Map();
```

Comment:
In Node most of the times using plain js objects/arrays > than using explicit types like Array<> or Map<>
They are easier to read and maintain most of the cases, unless you want to use something specific that those classes gives you.

A most common way to write this line could be:
```suggestion
 type Timer = {
  timeoutId: NodeJS.Timeout;
  startTime: number;
  duration: number;
};

type Timers = { [key: string]: Timer };

......

export default class TimerService {
    private timers: Timers = {};
```

No need to be changed now (lot of code already written like this) but it would be good to get used to the shortcuts so its easier to read

Reply:
I disagree. In that case the map object gives some useful functions that makes the code easier to maintain. Since Map is not a package it's a TS/JS object, I'll leave it as it is.

Reply:
Yes leave it as it is, but all the functionality you're using with Map in this class can be used with JS object.

JS objects has lots of functionalities, most probably we dont even know them all. 99% of cases no need for new Array/Map or others